develop/tutorials/articles/192-item-selector/01-selecting-entities-using-the-item-selector-api.markdown

```diff
@@ -0,0 +1,356 @@
+   Returns the following information of a file entry as a JSON object:
+   
+   -  `fileEntryId`: The ID of the selected file entry.
+   -  `groupId`: The group ID of the selected file entry.
+   -  `title`: The title of the selected file entry.
+   -  `url`: The URL of the selected FileEntry.
+   -  `uuid`: The UUID of the selected file entry.
+
+-  [URLItemSelectorReturnType](https://docs.liferay.com/portal/7.0/javadocs/modules/apps/collaboration/item-selector/com.liferay.item.selector.criteria.api/com/liferay/item/selector/criteria/URLItemSelectorReturnType.html)
+   Returns the URL of the entity as a `String`.
+
+-  [UUIDItemSelectorReturnType](https://docs.liferay.com/portal/7.0/javadocs/modules/apps/collaboration/item-selector/com.liferay.item.selector.criteria.api/com/liferay/item/selector/criteria/UUIDItemSelectorReturnType.html)
+   Returns the Universally Unique Identifier (UUID) of the entity as a `String`.
+
+These are the default criterion and return types available to the Item 
+Selector. However, applications can create their own criteria and return types 
+to select new types of entities. This is covered in more detail in the 
+[Creating Custom Entities for the Item Selector](/develop/tutorials/-/knowledge_base/7-0/creating-custom-entities-for-the-item-selector) 
+tutorial.
+
+Once you know the kind of element you want to select and the return type you
+want for the selected element, you can use the corresponding classes for the 
+item selector.
+
+For example, imagine that you want to select images and you expect the item 
+selector to return the URL of the selected image, then you can use the following 
+classes (both of them in the `item-selector-criteria-api` module):
+
+    com.liferay.item.selector.criteria.image.criterion.ImageItemSelectorCriterion
+
+    com.liferay.item.selector.criteria.URLItemSelectorReturnType
+
+The Item Selector uses this information to decide which selection views should 
+be displayed based on the criteria and the return types. For example, if you use 
+`ImageItemSelectorCriterion`, it will display the image selection views, while 
+if you use `VideoItemSelectorCriterion`, it will display the video selection 
+views.
+
+Now that you know how to specify the kind of element and return type that you 
+want to use for the Item Selector, you can learn how to use that information to 
+create a URL to open the Item Selector Dialog next.
+
+## Obtaining the URL for the Item Selector Views [](id=obtaining-the-url-for-the-item-selector-views)
+
+In order to use the Item Selector you'll need to obtain a URL to invoke it. This 
+URL is used to provide the proper views for the selected entity.
+
+The URL is generated by the Item Selector OSGi Component, so you'll need to 
+import the Item Selector OSGi Component into your class to use it.
```

Comment:
Hey @jhinkey since the ItemSelector is an OSGi component, it can only be used in classes that are components (unless it's received in a constructor). 

I think that the general case would be to obtain the itemSelector in a *MVCRenderCommand OSGi component using the @Reference notation, and then, either pass the url or the whole item selector object to the jsp or the template (freemarker, velocity, etc).
The other option that is also very common is to obtain the ItemSelector in a *ConfigContributor.java OSGi component. Then, the item selector url will be included in the json object that will be interpreted by the editor.